Cargo.toml

```diff
@@ -13,22 +13,26 @@ rust-version = "1.60.0"
 
 # The async runtime features mirror those of `zbus` for compatibility.
 [features]
-default = ["async-io"]
+default = ["async-io", "rs-crypto"]
 async-io = ["zbus/async-io"]
 tokio = ["zbus/tokio"]
+rs-crypto = ["dep:aes", "dep:block-modes", "dep:sha2", "dep:hkdf"]
+vendored-openssl = ["openssl", "openssl/vendored"]
```

Comment:
Theoretically, wouldn't this current feature set definition be a breaking change for anyone using the `tokio` feature on `main`? I tagged version 3.0, and haven't released it yet, but I'm imagining this scenario:
```toml
secret-service = { version = "3.0", default-features = false, features = ["tokio"] }
```

With the above, and assuming this was released as 3.1, Tokio users would then be left with a compilation error after upgrading because they no longer have a cryptography backend in a semver compatible update. A crate that I know has a very similar issue to this is `sqlx`, who has chosen [to create multiple feature combos](https://github.com/launchbadge/sqlx/blob/main/Cargo.toml#L85-L89) over both their async runtime and cryptography (TLS) backends. What are your thoughts on this?

Reply:
Hmm. For sqlx, that seems useful since they have multiple nested packages which they want a single feature flag to control. For secret-service-rs, that's not the case, and you're right that someone just throwing out `default-features = false` would result in a compilation error. Which is not great, but without the referenced issue being fixed I'm not sure there are a lot of alternatives.

Reply:
Sorry for the late reply, I have been sitting on this one trying to think of an alternative but I've come up with no good answers 😞. I've reached out to some people at work to see if there's any other ideas. Otherwise, I think the only option is to keep this as-is and change the documentation for the 3.0 release to tell people about the new requirement.

Reply:
Sounds good, and no rush 🙂

I will also be off the grid for two weeks starting tomorrow, so any replies/updates will be delayed...

Reply:
We discussed this a bit internally and the best conclusion a group of us were able to come up with is doing this SQLx-style with a feature for each of the 4 dependency combinations:

```toml
[features]
crypto-rust = ["<snip>"]
crypto-openssl = ["dep:openssl"]

rt-async-io-cryoto-rust = ["zbus/async-io", "crypto-rust"]
rt-async-io-crypto-openssl = ["zbus/async-io", "crypto-openssl"]

rt-tokio-crypto-rust = ["zbus/tokio", "crypto-rust"]
rt-tokio-crypto-openssl = ["zbus/tokio", "crypto-openssl"]
```

The above would be combined with some `compile_error!` statements to make sure selected feature sets made sense and to lead users into selecting one. There would no longer be a default either. What are your thoughts on this?

Reply:
I think that makes sense -- would you like me to try and implement that in this PR, or would you like to do it while merging?

Reply:
If you wouldn't mind, I'd appreciate it trying to happen in this PR.

Reply:
Done! Unfortunately, without selecting a default feature, the user gets a zbus compilation error first since that's compiled before the secret service package 😬 

<img width="737" alt="image" src="https://user-images.githubusercontent.com/2230985/201446814-67462ac1-e0ab-4df2-9040-39e03de095b2.png">

Reply:
I think that's fine for now 🤷. I can try to add more documentation to the README about this as well to make the happy path recovering from that error better.

Reply:
cool 🙂  give me a ping if there's anything else you see this PR needs to be mergable

Reply:
FWIW, have you pushed your most recent commits up? I haven't seen anything on GitHub beyond the original changes 😅.

Reply:
d'oh, I was pushing to the wrong place